lua/neogit/config.lua

```diff
@@ -1,157 +1,493 @@
+        or not validate_type(key_binds, "mappings.finder." .. vim.inspect(command), "table")
+      then
+        return
+      end
+
+      if not vim.tbl_contains(valid_finder_commands, command) then
+        err(
+          string.format(
+            "Expected a valid finder command, got '%s'. Valid finder commands: { %s }",
+            command,
+            table.concat(valid_finder_commands, ", ")
+          )
+        )
+      end
+
+      for _, key in ipairs(key_binds) do
+        validate_type(key, string.format("mappings.finder.%s -> %s", command, vim.inspect(key)), "string")
+      end
+    end
+
+    if validate_type(config.mappings.finder, "mappings.finder", "table") then
+      for command, key_binds in pairs(config.mappings.finder) do
+        validate_finder_map(command, key_binds)
+      end
+    end
+
+    -- Validate mappings.status
+    local valid_status_commands = {
+      "Close",
+      "InitRepo",
+      "Depth1",
+      "Depth2",
+      "Depth3",
+      "Depth4",
+      "Toggle",
+      "Discard",
+      "Stage",
+      "StageUnstaged",
+      "StageAll",
+      "Unstage",
+      "UnstageStaged",
+      "DiffAtFile",
+      "CommandHistory",
+      "Console",
+      "RefreshBuffer",
+      "GoToFile",
+      "VSplitOpen",
+      "SplitOpen",
+      "TabOpen",
+      "HelpPopup",
+      "DiffPopup",
+      "PullPopup",
+      "RebasePopup",
+      "MergePopup",
+      "PushPopup",
+      "CommitPopup",
+      "LogPopup",
+      "RevertPopup",
+      "StashPopup",
+      "CherryPickPopup",
+      "BranchPopup",
+      "FetchPopup",
+      "ResetPopup",
+      "RemotePopup",
+      "GoToPreviousHunkHeader",
+      "GoToNextHunkHeader",
+    }
+    if validate_type(config.mappings.status, "mappings.status", "table") then
+      for command, key in pairs(config.mappings.status) do
+        if
+          validate_type(command, "mappings.status -> " .. vim.inspect(command), "string")
+          and validate_type(key, "mappings.status." .. command, "string")
+        then
+          if not vim.tbl_contains(valid_status_commands, command) then
+            err(
+              string.format(
+                "Expected a valid status command, got '%s'. Valid finder commands: { %s }",
+                command,
+                table.concat(valid_status_commands, ", ")
+              )
+            )
+          end
+        end
+      end
+    end
+  end
+
+  if validate_type(config, "base config", "table") then
+    validate_type(config.disable_hint, "disable_hint", "boolean")
+    validate_type(config.disable_context_highlighting, "disable_context_highlighting", "boolean")
+    validate_type(config.disable_signs, "disable_signs", "boolean")
+    validate_type(config.disable_commit_confirmation, "disable_commit_confirmation", "boolean")
+    validate_type(config.disable_builtin_notifications, "disable_builtin_notifications", "boolean")
+    validate_type(config.use_telescope, "use_telescope", "boolean")
```

Comment:
`use_telescope` - This got removed today

Reply:
Removed it from my branch, including the tests.